_drafts/hw_actors.md

```diff
@@ -1,18 +1,21 @@
 ---
 title: The hardware-implemented actor model inside modern MCUs
-description: Description of how interrupt controllers may be used as simple schedulers
+description: Description of how interrupt controllers may function as simple schedulers
 author: ramanf
 image:
 ---
 
-<!-- A little bit of background of why the reader should read this post. -->
+<!-- A little background on why readers should read this post. -->
 
-Many embedded RTOS now have at least a few thousands lines of code, so a simple lightweight framework is often needed for embedded devices. Since ARM is the most popular embedded CPU and almost every ARM processor has a hardware scheduler, I think it would be interesting to make a compact framework utilizing these features. As far as I know, the most popular approach to RTOS-less event-driven systems is super-loop consuming events that are produced by interrupts. This article discusses an alternative approach which seems to be more efficient. It has less overhead than RTOS and also brings some benefits to event-driven systems.
+Many embedded RTOS now have at least a few thousand lines of code, so a simple, lightweight framework is often needed for embedded devices. Since ARM is the most popular embedded CPU and almost every ARM processor has a hardware scheduler, it would be interesting to make a compact framework utilizing these features. The most popular approach to RTOS-less event-driven systems is super-loop consuming events produced by interrupts. This article discusses an alternative, more efficient approach in some circumstances. It has less overhead than RTOS and also benefits event-driven systems.
```

Comment:
I'm still not fully sold on mentioning the number of lines of code as an indicator of complexity. Maybe we could rephrase this along the lines of:

```
Using an RTOS is often a tradeoff between the ease of decomposing tasks, with the complexity of the scheduler itself. There exists a middle ground between highly complex systems that may require an RTOS, and simpler ones that can be easily modeled using a super loop.
```